code/game/objects/items/stacks/medical.dm

```diff
@@ -85,12 +95,21 @@
 		user.visible_message("<span class='notice'>[user] starts to apply [src] on [user.p_them()]self...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
 		if(!do_mob(user, M, self_delay))
 			return
+		//After the do_mob to ensure metabolites have had time to process at least one tick. 
+		if(reagent && (C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/styptic_powder) || C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/silver_sulfadiazine)))
+			to_chat(user, "<span class='warning'>That stuff really hurt! You can't bring yourself to apply [src] to your wounds again yet, maybe someone else can help put it on for you.</span>")
+			return
 
 	user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")
-
-	if(affecting.heal_damage(heal_brute, heal_burn) || stop_bleeding)
+	if(reagent)
+		reagents.reaction(M, PATCH, affecting = affecting)
+		M.reagents.add_reagent_list(reagent) //Stack size is reduced by one instead of actually removing reagents from the stack.
```

Comment:
there should be a code that reagent in a pack is removed by applied amount, so that you won't get the reagents from grinding those after using those.
so, `remove_reagent(medicine, REAGENT_AMOUNT_PER_ITEM/amount)` would work.

Reply:
Actually, this is how I initially did it and even had it working appropriately. I changed it to the system I'm using now because it took far less code/steps to achieve the desired results: 

It introduced a plethora of bugs when it comes to splitting stacks of bruise packs and ointments: When you split a stack an entirely new object is initialized, and reagents are not split. 

Say you have a stack of six ointment, which would be 120u of ointment. You split that into a separate stacks of 3 ointment each? You still have 120 ointment in the first stack, and have now initialized a new stack of ointment which will be given whatever the initial value of reagents was. There were two ways to solve this, and both were way more convoluted and introduced much worse snowflakey code than the solution I went with I feel like. 

In addition to the above: when trying to transfer reagents an entire stack of five is treated as one object when it comes to transferring reagents, so I'd be reducing the stack size by 1 *and also* manually reducing one stack's worth of reagents from the stack to ensure the stack maintained the correct amount of reagents.

There's so many ways for this to go wrong and bug out that I probably didn't encounter when I got it working too.